datafusion-examples/examples/dataframe.rs

```diff
@@ -109,3 +129,135 @@ async fn example_read_csv_file_with_schema(file_path: &str) -> DataFrame {
     // Register a lazy DataFrame by using the context and option provider
     ctx.read_csv(file_path, csv_read_option).await.unwrap()
 }
+
+/// This example demonstrates how to use the to_date series
+/// of functions in the DataFrame API
```

Comment:
Some of the consolidated examples had both SQL and dataframe examples, so I split such examples into dataframe.rs and sql.rs